File: python_chunking/test_files/Calculator.java
/**
 * Java Calculator class for testing chunking
 */
import java.util.ArrayList;
import java.util.List;

public class Calculator {
    private double result;
    private List<String> history;
    
    /**
     * Initialize calculator with result 0
     */
    public Calculator() {
        this.result = 0.0;
        this.history = new ArrayList<>();
    }
    
    /**
     * Add two numbers and return result
     * @param x First number
     * @param y Second number
     * @return Sum of x and y
     */
    public double add(double x, double y) {
        double result = x + y;
        this.history.add(String.format("Added %.2f + %.2f = %.2f", x, y, result));
        return result;
    }
    
    /**
     * Subtract y from x and return result
     * @param x First number
     * @param y Second number
     * @return Difference of x and y
     */
    public double subtract(double x, double y) {
        double result = x - y;
        this.history.add(String.format("Subtracted %.2f - %.2f = %.2f", x, y, result));
        return result;
    }
    
    /**
     * Multiply two numbers and return result
     * @param x First number
     * @param y Second number
     * @return Product of x and y
     */
    public double multiply(double x, double y) {
        double result = x * y;
        this.history.add(String.format("Multiplied %.2f * %.2f = %.2f", x, y, result));
        return result;
    }
    
    /**
     * Divide x by y and return result
     * @param x Dividend
     * @param y Divisor
     * @return Quotient of x and y
     * @throws ArithmeticException if y is zero
     */
    public double divide(double x, double y) throws ArithmeticException {
        if (y == 0) {
            throw new ArithmeticException("Cannot divide by zero");
        }
        double result = x / y;
        this.history.add(String.format("Divided %.2f / %.2f = %.2f", x, y, result));
        return result;
    }
    
    /**
     * Get calculation history
     * @return Copy of calculation history
     */
    public List<String> getHistory() {
        return new ArrayList<>(this.history);
    }
    
    /**
     * Clear calculation history
     */
    public void clearHistory() {
        this.history.clear();
    }
    
    /**
     * Main method to test calculator
     * @param args Command line arguments
     */
    public static void main(String[] args) {
        Calculator calc = new Calculator();
        
        // Test basic operations
        System.out.println("Testing Calculator:");
        System.out.println("5 + 3 = " + calc.add(5, 3));
        System.out.println("10 - 4 = " + calc.subtract(10, 4));
        System.out.println("6 * 7 = " + calc.multiply(6, 7));
        System.out.println("15 / 3 = " + calc.divide(15, 3));
        
        // Show history
        System.out.println("\nCalculation History:");
        for (String entry : calc.getHistory()) {
            System.out.println("  " + entry);
        }
    }
}

File: python_chunking/test_files/Calculator.py
"""
Python Calculator class for testing chunking
"""
class Calculator:
    """A simple calculator class"""
    
    def __init__(self):
        """Initialize calculator with result 0"""
        self.result = 0
        self.history = []
    
    def add(self, x, y):
        """Add two numbers and return result"""
        result = x + y
        self.history.append(f"Added {x} + {y} = {result}")
        return result
    
    def subtract(self, x, y):
        """Subtract y from x and return result"""
        result = x - y
        self.history.append(f"Subtracted {x} - {y} = {result}")
        return result
    
    def multiply(self, x, y):
        """Multiply two numbers and return result"""
        result = x * y
        self.history.append(f"Multiplied {x} * {y} = {result}")
        return result
    
    def divide(self, x, y):
        """Divide x by y and return result"""
        if y == 0:
            raise ValueError("Cannot divide by zero")
        result = x / y
        self.history.append(f"Divided {x} / {y} = {result}")
        return result
    
    def get_history(self):
        """Get calculation history"""
        return self.history.copy()
    
    def clear_history(self):
        """Clear calculation history"""
        self.history.clear()


def main():
    """Main function to test calculator"""
    calc = Calculator()
    
    # Test basic operations
    print("Testing Calculator:")
    print(f"5 + 3 = {calc.add(5, 3)}")
    print(f"10 - 4 = {calc.subtract(10, 4)}")
    print(f"6 * 7 = {calc.multiply(6, 7)}")
    print(f"15 / 3 = {calc.divide(15, 3)}")
    
    # Show history
    print("\nCalculation History:")
    for entry in calc.get_history():
        print(f"  {entry}")


if __name__ == "__main__":
    main()
